daemon/containerd/image_builder.go

```diff
@@ -328,6 +328,35 @@ func (rw *rwlayer) Commit() (_ builder.ROLayer, outErr error) {
 		}
 	}()
 
+	// Temporarily unmount the layer, required by the containerd windows snapshotter.
+	// The windowsfilter graphdriver does this inside its own Diff method. We'll put it
+	// back afterwards.
+
+	// TODO: Do we actually need to put it back aterwards?
+	// The only place that calls this in-tree is (b *Builder) exportImage and
+	// that is called from the end of (b *Builder) performCopy which has a
+	// `defer rwLayer.Release()` pending.
```

Comment:
Thinking about the WCOW snapshotter implementation in containerd, once you Commit a snapshot, any future changes you make to it will leave it in an inconsistent state. This is because (unlike the graphdriver), Commit updates the snapshot _in-place_ from a RWLayer to a ROLayer. The RWLayer data (sandbox.vhdx) is kept around so that 'Diff' works later, but it's actually impossible from the API to get a new read-write reference to that layer, as the old snapshot key is invalidated, and the new snapshot key can only be used as a parent for new layers.

We get around this here by capturing the mount data _before_ Commit, but I realise now that's more hacky than I intended, as due to the difference with the graphdriver behaviour, anything touching that data will make future `Diff` calls inconsistent.

So perhaps `rwlayer.Commit()` should imply `Release()`? That's how it's currently used now, anyway.

Reply:
According to the interface Godoc that's true for the `Commit` method in general, not only WCOW:

```
	// Commit captures the changes between key and its parent into a snapshot
	// identified by name.  The name can then be used with the snapshotter's other
	// methods to create subsequent snapshots.
	//
	// A committed snapshot will be created under name with the parent of the
	// active snapshot.
	//
	// After commit, the snapshot identified by key is removed.
	Commit(ctx context.Context, name, key string, opts ...Opt) error
```

So the current Linux implementation already assumes that the layer won't be used anymore after `Commit`.
IMO, we can just skip the remount. It would be nice if we could keep the API consistent and make the graphdriver's `Commit` also release the layer, but given the single usage and the fact that we're probably not going to have any new code that would use this I don't think it's strictly needed.

Reply:
I pushed a commit that removes the remount.

Reply:
I was thinking about rebasing this PR the other day, so thank you for this.

If more non-trivial work is needed for this PR, and I'm doing it, do you have any concerns if I squash that "remount removal" commit into the commit it's modifying ("Temporarily unmount RWLayer during Commit")?

Reply:
Nope, I did it as a separate commit to not mix my changes with your original PR and make it more visible was was changed. Will squash it now.